### PR TITLE
Add documentation and remove dead code.

### PR DESCRIPTION
**Pull request by AureliaDolo**

### CODE REVIEW

**Review by AurelienFT — DISMISSED** [review]
Good documentation !

**Comment by AurelienFT** [comment]
What is the rule about capital letter at the start of a documentation line ?

**Comment by AureliaDolo** [comment]
> What is the rule about capital letter at the start of a documentation line ?

maybe we can decide something here #2287 ? 

I have to admit that I am not consistent in formatting.

Once @yvan-sraka tried to format everything

File: massa-consensus-worker/src/consensus_worker.rs [hunk 1]
```diff
@@ -1161,6 +1190,21 @@ impl ConsensusWorker {
         Ok(())
     }
 
+    /// call me if the block database changed
```
Comment (adrien-zinger) [inline massa-consensus-worker/src/consensus_worker.rs:1193]:
```suggestion
    /// Processing of final blocks, pruning and producing endorsement.
    /// Called at each block tick and each time protocol signal a new block or
    /// a new block header.
```

File: massa-consensus-worker/src/tests/scenario_block_creation.rs [hunk 2]
```diff
@@ -73,7 +73,50 @@ async fn test_genesis_block_creation() {
...
+///
+/// ### Initialization
+/// Following blocks are sent through a protocol event to consensus right at the beginning. They all have best parents as parents.
+/// * block at slot(1,0) with operation addr 2 buys 1 roll
+/// * block at slot( period per cycle, 0)
+/// * block at slot( period per cycle, 1)
+/// * block at slot( period per cycle + 1, 0)
+/// * block at slot( period per cycle + 1, 1)
+/// * block at slot( period per cycle + 2, 0)
+/// * block at slot( period per cycle + 2, 0)
```
Comment (adrien-zinger) [inline massa-consensus-worker/src/tests/scenario_block_creation.rs:103]: Did you mean ... ?
```suggestion
/// * block at slot( period per cycle + 2, 1)
```

File: massa-client/src/cmds.rs [hunk 3]
```diff
@@ -23,6 +23,9 @@ use std::process;
 use strum::{EnumMessage, EnumProperty, IntoEnumIterator};
 use strum_macros::{Display, EnumIter, EnumMessage, EnumProperty, EnumString};
 
+/// All the client commands
+/// the order they are defined is the order they are displayed in so be careful
+/// Maybe it would be worth renaming some of them for consistency
```
[the remark this answers is not in the document]
Comment (AureliaDolo) [reply massa-client/src/cmds.rs:28]: I tried to keep commands that go through private and public api separated and also wallet specific commands last. But really it's just a mess and as there is no convention on naming, alphabetical order may be confusing

Maybe some spec, or some discussion on what we want to do with the client would be useful @damip @Kryptopaid (the russian one looks shinny)

File: massa-consensus-worker/src/tests/scenario_block_creation.rs (hunk 2 above)
Comment (AureliaDolo) [reply massa-consensus-worker/src/tests/scenario_block_creation.rs:103]: I don't remember x)
I just copy pasted the scenario as described in the issue
Maybe it's an issue with the test ?

**Comment by AureliaDolo** [comment]
I think it would be nice to merge this quite quickly to avoid to much conflicts

File: massa-client/src/cmds.rs [hunk 4]
```diff
@@ -185,27 +188,35 @@ pub enum Command {
...
+/// print a yellow warning
 macro_rules! client_warning {
     ($e:expr) => {
         println!("{}: {}", style("WARNING").yellow(), $e)
     };
 }
 
-#[derive(Serialize)]
+/// Used to have a shinny json output
+/// Really I'm not sure why this exist
```
Comment (AurelienFT) [inline massa-client/src/cmds.rs:212]: We must sort this out before merging. We can have this type of comment in our codebase.

File: massa-client/src/cmds.rs [hunk 5]
```diff
@@ -219,10 +230,13 @@ impl Display for ExtendedWalletEntry {
...
 }
 
-#[derive(Serialize)]
+/// Aggregation of the local, with some useful information as the balance, etc
+/// to be printed by the client.
+#[derive(Debug, Serialize)]
 pub struct ExtendedWallet(Map<Address, ExtendedWalletEntry>);
 
 impl ExtendedWallet {
+    /// Just reorganize everything into an extended wallet
```
Comment (AurelienFT) [inline massa-client/src/cmds.rs:239]:
```suggestion
    /// Reorganize everything into an extended wallet
```

File: massa-models/src/api.rs [hunk 6]
```diff
@@ -139,34 +156,51 @@ impl std::fmt::Display for RollsInfo {
...
     pub module: Vec<u8>,
+    /// datastore
     pub datastore: Map<Hash, Vec<u8>>,
 }
 
 impl std::fmt::Display for SCELedgerInfo {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         writeln!(f, "\tBalance: {}", self.balance)?;
-        // I choose not to display neither the module nor the datastore because bytes
+        // I choose not to display neither the module nor the datastore because bytes eh
```
Comment (AurelienFT) [inline massa-models/src/api.rs:173]: What "eh" means ?

File: massa-models/src/composite.rs [hunk 7]
```diff
@@ -6,45 +6,67 @@ use massa_signature::{PublicKey, Signature};
...
     Discarded,
-    Stored,
 }
 
+/// Status in which an operation can be
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub enum OperationSearchResultStatus {
+    /// in pool
     Pending,
+    /// in a block, the block being in OperationSearchResultBlockStatus
```
Comment (AurelienFT) [inline massa-models/src/composite.rs:29]: Use the link method : https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html

**Review by AurelienFT — COMMENTED** [review]
In a more general way, I still don't understand your personal capital rule. Sometimes you have it on fields sometimes not. A lot of time the first line of the doc don't have it but the second one does it's really strange. I understand that there is no global rule but in this context I think the language and uniformity rule should apply.

File: massa-models/src/api.rs [hunk 8]
```diff
@@ -68,16 +68,25 @@ impl std::fmt::Display for NodeStatus {
...
     pub in_blocks: Vec<BlockId>,
+    /// true if the operation is final (ie in a final block)
     pub is_final: bool,
+    /// the operation itself
     pub operation: SignedOperation,
 }
 
 impl OperationInfo {
+    /// extend an operation info with another one
+    /// There is not check to see if the id and operation are indeed the same
```
[the remark this answers is not in the document]
Comment (AureliaDolo) [reply massa-models/src/api.rs:89]: I don't understand your comment

[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-models/src/api.rs:89]: Your comment is useful but maybe I missed it. It's that a problem that there is no check or it's just an information ?

[the remark this answers is not in the document]
Comment (AureliaDolo) [reply massa-models/src/api.rs:89]: It's just an information. If you look at the various check_xxx and note_xxxx_from_node (and maybe the deserialization ones too) the is a description of the checks that are performed and at which level. 

For example, if you take a block, you'll check the parent count at deserialization level only whereas you'll check if the parents are compatible together only at consensus level.

**Review by yvan-sraka — DISMISSED** [review]

File: massa-models/src/api.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-models/src/api.rs:89]: Ok thanks for clarifications :) !

**Review by yvan-sraka — DISMISSED** [review]

**Review by AurelienFT — COMMENTED** [review]
Reviewed half of the files. Still have comments open on my previous review also.

File: massa-network-exports/src/establisher.rs [hunk 9]
```diff
@@ -3,9 +3,13 @@
 mod types {
     use crate::test_exports::mock_establisher;
 
+    /// read half
```
Comment (AurelienFT) [inline massa-network-exports/src/establisher.rs:6]: I think it should be more explicit. Same for write half.

File: massa-network-exports/src/establisher.rs [hunk 10]
```diff
@@ -17,9 +21,13 @@ mod types {
         time::timeout,
     };
 
+    /// read half
```
Comment (AurelienFT) [inline massa-network-exports/src/establisher.rs:24]: Same as above

File: massa-network-exports/src/test_exports/mock_establisher.rs [hunk 11]
```diff
@@ -10,11 +10,14 @@ use tokio::time::timeout;
 
 const MAX_DUPLEX_BUFFER_SIZE: usize = 1024;
 
+/// read half
```
Comment (AurelienFT) [inline massa-network-exports/src/test_exports/mock_establisher.rs:13]: I think it should be more explicit. Same for write half.

File: massa-api/src/lib.rs [hunk 12]
```diff
@@ -34,31 +34,52 @@ mod error;
 mod private;
 mod public;
 
+/// Public api component
 pub struct Public {
+    /// link to the consensus component
     pub consensus_command_sender: ConsensusCommandSender,
```
[the remark this answers is not in the document]
Comment (AureliaDolo) [reply massa-api/src/lib.rs:40]: Except it has a bad name, from the outside, we don't send commands, we just call consensus command sender's methods

File: massa-bootstrap/src/lib.rs [hunk 13]
```diff
@@ -1,12 +1,19 @@
 // Copyright (c) 2022 MASSA LABS <info@massa.net>
-
+//! Bootstrap crate
+//!
+//! At start up is now is after genesis timestamp,
```
[the remark this answers is not in the document]
Comment (AureliaDolo) [reply massa-bootstrap/src/lib.rs:4]: Me neither x), I'll take a look

[the remark this answers is not in the document]
Comment (AureliaDolo) [reply massa-bootstrap/src/lib.rs:4]: It was a typo: is instead of if

**Comment by AureliaDolo** [comment]
@AurelienFT for the read/write half, I really don't know what to add

File: massa-bootstrap/src/lib.rs (hunk 13 above)
[the remark this answers is not in the document]
Comment (AureliaDolo) [reply massa-bootstrap/src/lib.rs]: And I added a comma for more clarity

**Comment by AurelienFT** [comment]
> @AurelienFT for the read/write half, I really don't know what to add

On my mind when we say "half" it's half of something it can't just be an half. I think something like "read half of the socket" or whatever it is.

File: massa-api/src/lib.rs (hunk 12 above)
[the remark this answers is not in the document]
Comment (AurelienFT) [reply massa-api/src/lib.rs:40]: Mhhh ok I think it can be explicited by saying that his methods will interact directly with the consensus worker.

**Review by AurelienFT — COMMENTED** [review]
Very nice documentation ! Still some points to improve on consistency etc but need to be discussed before. I don't approve because we shouldn't merge this before the storage but will make it after.

**Comment by AureliaDolo** [comment]
> Very nice documentation ! Still some points to improve on consistency etc but need to be discussed before. I don't approve because we shouldn't merge this before the storage but will make it after.

Storage is in the bors :)

**Comment by AurelienFT** [comment]
> Storage is in the bors :)

After conflict resolution I will approve

**Comment by AureliaDolo** [comment]
@adrien-zinger @gterzian I squashed all my commits after merging with main.

**Review by AurelienFT — APPROVED** [review]

**Review by yvan-sraka — APPROVED** [review]

**Review by adrien-zinger — APPROVED** [review]

**Comment by AureliaDolo** [comment]
bors merge

**Pull request merged**